src/Dapr.Client/DaprClient.cs

```diff
@@ -939,6 +940,286 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
+        /// <param name="encryptionOptions">Options informing how the encryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract IAsyncEnumerable<byte[]> EncryptStreamAsync(string vaultResourceName, Stream plaintextStream, string keyName,
+            EncryptionOptions encryptionOptions, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Decrypts the specified ciphertext bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="ciphertextBytes">The bytes of the ciphertext value to decrypt.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the decryption operation.</param>
+        /// <param name="options">Options informing how the decryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of decrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<ReadOnlyMemory<byte>> DecryptAsync(string vaultResourceName, ReadOnlyMemory<byte> ciphertextBytes, string keyName, DecryptionOptions options,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Decrypts the specified ciphertext bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="ciphertextStream">The stream containing the bytes of the ciphertext value to decrypt.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the decryption operation.</param>
+        /// <param name="options">Options informing how the decryption operation should be configured.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of decrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<ReadOnlyMemory<byte>> DecryptAsync(string vaultResourceName, Stream ciphertextStream, string keyName, DecryptionOptions options,
```

Comment:
I wonder if we could start with just two basic forms of each method, one which accepts a stream and returns an async enumerator (the "efficient" case) and one which accepts a whole buffer and returns a whole buffer, and see how that goes with users.  I find the additional "accepts a stream but returns a whole buffer" a bit unusual and I think makes the naming complicated.

Reply:
I think this is a good suggestion. That's what we do in the JS SDK too.

Reply:
It's a fair complaint - I left it in there since the implementation method itself accepts a `Stream` but returns a `ReadOnlyMemory<byte>` and it felt weird to have a method that did precisely that but which wasn't available to the developer to use themselves. Nonetheless, removed from the Dapr client and swapped the visibility in the gRPC implementation to private for both encrypt and decrypt.